src/OrleansServiceBus/Providers/Streams/EventHub/EventHubCheckpoint.cs

```diff
@@ -3,17 +3,14 @@
 using System.Threading.Tasks;
 using Microsoft.ServiceBus.Messaging;
 using Orleans.AzureUtils;
+using Orleans.Streams;
 
 namespace Orleans.ServiceBus.Providers
 {
     /// <summary>
     /// This class stores EventHub partition checkpoint information (a partition offset) in azure table storage.
-    /// TODO: Use dependency injection to fill this behavior.
-    /// This class stores data into table storage, which will not be ideal for all users.  Further, it introduces
-    ///   a dependency on azure storage to this assembly.  Ideally this should be an injectable behavior, rather
-    ///   than a concrete class.
     /// </summary>
-    internal class EventHubPartitionCheckpoint
+    public class EventHubCheckpoint : IStreamQueueCheckpoint<string>
```

Comment:
`AzureTableEventHubCheckpointer`? Now we can actually replace it with something different